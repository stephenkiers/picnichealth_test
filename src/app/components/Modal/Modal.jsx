import React, {Component} from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'

class Modal extends Component {

    render() {
        return (
            <div className="modal">
                <div className="modal-contents">
                    <div className="modal-body">
                        {this.props.children}
                    </div>
                </div>
                <div className="modal-bg"
                     onClick={this.props.closeModal}
                />
            </div>

        )
    }
}
Modal.defaultProps = {
};
Modal.propTypes = {
    closeModal: PropTypes.func.isRequired,
}

export default Modal
