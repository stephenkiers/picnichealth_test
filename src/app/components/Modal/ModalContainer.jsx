import React, {Component} from 'react';
import { connect } from 'react-redux';
import Modal from "./Modal";
import {reduxAppStateSetModalId} from "../../AppState/actions";

class ModalContainer extends Component {

    componentDidMount() {
        window.addEventListener("keydown", this.handleEscapeKeyPress.bind(this), false);
    }
    componentWillUnmount() {
        window.removeEventListener("keydown", this.handleEscapeKeyPress.bind(this), false);
    }
    handleEscapeKeyPress(e) {
        if(e.keyCode === 27) {
            this.props.closeModal();
        }
    }

    render() {
        return (
            <Modal
                closeModal={this.props.closeModal}
            >
                {this.props.children}
            </Modal>
        )
    }
}
ModalContainer.defaultProps = {
};
ModalContainer.propTypes = {
}

const mapStateToProps = (state, ownProps) => ({
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    closeModal() {
        dispatch(reduxAppStateSetModalId(undefined));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalContainer);
