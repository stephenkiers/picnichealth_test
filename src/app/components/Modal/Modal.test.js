import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Modal from './Modal';

test('Modal displays correct', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Modal
            closeModal={() => {}}
        />
    );
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot();
});