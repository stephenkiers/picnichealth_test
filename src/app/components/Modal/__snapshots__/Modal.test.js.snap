// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal displays correct 1`] = `
<div
  className="modal"
>
  <div
    className="modal-contents"
  >
    <div
      className="modal-body"
    />
  </div>
  <div
    className="modal-bg"
    onClick={[Function]}
  />
</div>
`;
